### ISSUE TITLE
Configure number of QuickCheck iterations

### ISSUE THREAD

**Issue by koalaman** [sol/doctest#212]
Can I configure the number of QuickCheck iterations somehow?

I have a 1300 properties that I want to check for truth, which don't vary with input and therefore only need to be executed once:

    -- prop> worksFor "foo"
    -- prop> worksFor "bar"
    -- prop> worksFor "baz"

With `doctest`, each property appears to be checked many times, presumably QuickCheck's default 100. This happens even though the expression type is a pure `Bool`. 

Before `doctest` I could add a footer with `(quickCheckWithResult (stdArgs { maxSuccess = 1 }  ) )`  to only test each property once.

I don't particularly want to rewrite all the properties to fit the `>>>` paradigm better, or to add 1300 dummy lines of `-- True`. What are my options? 

I tried `doctests --help` and variations, but the binary appears to ignore command line flags.

**Comment by sol** [comment]
Upfront, 1300 properties sounds a lot!  Doctest is a tool for testing documentation, not code!  If these properties are not essential as user documentation and if you are actually testing your code here (in contrast to checking that the examples in your user documentation do actually work), then Doctest is not the ideal choice for your needs.

Reason: Doctest is slow.  For code testing there are much faster and better tools out there.  You could stick with plain QuickCheck or give [`hspec`](https://hspec.github.io/) a try (full disclosure, I'm the author of `hspec`).

---

> Can I configure the number of QuickCheck iterations somehow?

I believe this is currently not possible.

> With `doctest`, each property appears to be checked many times, presumably QuickCheck's default 100. This happens even though the expression type is a pure `Bool`.

For `Bool` QuickCheck will only run one test.  Try this in GHCi:

```
$ ghci
ghci> import Test.QuickCheck
ghci> quickCheck True
+++ OK, passed 1 test.
```

This should also be true for `doctest`.  If it is not please open a bug report with minimal steps on how to reproduce.

> I tried `doctests --help` and variations, but the binary appears to ignore command line flags.

`doctest` accepts the same command-line flags as `ghci`, for that reason adding additional flags is somewhat tricky.

**Comment by sol** [comment]
You can try `--fast` which should make it run faster.

**Comment by koalaman** [comment]
Wow, you're right.

Just to check whether or not I'm crazy, I just built an older version with QuickCheck 2.7.4 from ~4 years ago. It shows "+++ OK, passed 1 tests." individually like you demonstrate, but when run via the TemplateHaskell `runTests`, it shows "+++ OK, passed 100 tests." for the exact same property. I see that this issue has since been fixed.

Doctest actually looked great because the tests -- being comments -- don't contribute to compile time or binary size. ShellCheck currently struggles with that, and has some scripts that strip test code from source files to get around it. Doctest made this much cleaner and nicer, but slower than ghci+quickCheck. I'll look further into it, thanks!
